core/include/nes/core/ppu_registers.h

```diff
@@ -4,6 +4,55 @@
 
 namespace n_e_s::core {
 
+template <class TypeT>
+class Register {
+public:
+    constexpr Register() = default;
+    constexpr explicit Register(const TypeT value) : value_{value} {};
+
+    [[nodiscard]] constexpr bool operator==(const Register &) const = default;
+    [[nodiscard]] constexpr bool operator!=(const Register &) const = default;
```

Comment:
Defining == makes the compiler generate != for you (in C++20 and newer.)
>The inequality operator is automatically generated by the compiler if operator== is defined.
https://en.cppreference.com/w/cpp/language/operators

Reply:
Cool, didn't know that! I'll remove it